jena-extras/jena-serviceenhancer/src/test/java/org/apache/jena/sparql/service/enhancer/assembler/TestServiceEnhancerDatasetAssembler.java

```diff
@@ -115,4 +124,54 @@ public void testAssemblerMgmtSuccess() {
             Assert.assertEquals(1, ResultSetFormatter.consume(qe.execSelect()));
         }
     }
+
+    /**
+     * Test case to check that when {@code OpService('loop:', subOp)} is evaluated against an union default
+     * graph (UDG) enabled dataset then the subOp must be evaluated on the UDG too.
+     * This also applies to {@code cache:} and {@code bulk:}.
+     */
+    @Test
+    public void testAssemblerTdbUnionDefaultGraph() throws IOException {
```

Comment:
This test case uses the service enhancer with an active dataset that is a default union graph. The pattern inside the service clause is expected to be evaluated against the default union graph as well.